src/main/java/Chatbot.java

```diff
@@ -0,0 +1,268 @@
+                + "----------\n"
+                + "  " + tasks.getItem(index) + "\n"
+                + "----------";
+    }
+
+    /**
+     * Parse delete command.
+     * @param tokens The input tokens.
+     * @return The reply to the done command.
+     * @throws DukeException The command syntax error.
+     */
+    String parseDelete(String[] tokens) throws DukeException {
+        DukeException invalidIndex = 
+                new DukeException("DELETE requires an index in the list to mark.");
+        int index;
+        
+        if (tokens.length != 1) {
+            throw invalidIndex;
+        }
+
+        try {
+            index = Integer.parseInt(tokens[0]);
+        } catch (NumberFormatException e) {
+            throw invalidIndex;
+        }
+
+        if (index < 1 || index > tasks.getLength()) {
+            throw invalidIndex;
+        }
+
+        Item item = tasks.getItem(index);
+        tasks.deleteItem(index);
+        tasks.save(savePath);
+        return "Removed and never to be seen again.\n"
+                + "----------\n"
+                + "  " + item + "\n"
+                + "----------\n"
+                + "You have " + tasks.getLength() + " tasks in the list. :(";
+    }
+
+    /**
+     * Parsing the user's input and returning a reply.
+     * @param message The user input message.
+     * @return Chatbot's reply to the message.
+     */
+    String parse(String[] tokens) throws DukeException {
+        String command = tokens[0].toUpperCase();
+
+        if (command.length() < 1) {
+            throw new DukeException("I have made you speechless.");
+        }
+
+        tokens = Arrays.copyOfRange(tokens, 1, tokens.length);
+
+        switch (command) {
+        case "TODO":
+            return parseTodo(tokens);
+        case "DEADLINE":
+            return parseDeadline(tokens);
+        case "EVENT":
+            return parseEvent(tokens);
+        case "LIST":
+            return "This is your useless list.\n"
+                    + tasks.toString();
+        case "DONE":
+            return parseDone(tokens);
+        case "DELETE":
+            return parseDelete(tokens);
+        case "BYE":
+            return "Bye, see you never!";
+        default:
+            throw new DukeException("What's this? You're wasting my time!");
+        }
+    }
+
+    /**
+     * Reply the user input message and displaying error as reply.
+     * @param message The input message.
+     * @return The reply to the message.
+     */
+    String reply(String message) {
+        try {
+            String[] tokens = message.split(" ");
+            return parse(tokens);
+        } catch (DukeException e) {
+            return e.getMessage();
+        }
+    }
+
+    /**
+     * Display what the chatbot is saying.
+     * @param message The message the chatbot is saying.
+     */
+    void say(String message) {
+        System.out.println(String.format("%s: %s\n", name, message));
+    }
+}
```

Comment:
all of your methods here do not have access public/private modifier. Would be good to indicate explicitly for methods whether it is public or private although no access modifier does have it own "privacy".